src/dotnet-interactive/Utility/KernelDiagnostics.cs

```diff
@@ -3,20 +3,20 @@
 
 using System;
 using System.Diagnostics;
+using Microsoft.DotNet.Interactive.Formatting;
 using Pocket;
 
 namespace Microsoft.DotNet.Interactive;
 
 internal static class KernelDiagnostics
 {
-
     [DebuggerStepThrough]
     public static T LogCommandsToPocketLogger<T>(this T kernel)
         where T : Kernel
     {
         kernel.AddMiddleware(async (command, context, next) =>
         {
-            using var _ = Logger.Log.OnEnterAndExit($"Command: {command.ToString().Replace(Environment.NewLine, " ")}");
+            using var _ = Logger.Log.OnEnterAndExit($"Command: {command.ToDisplayString("text/log")}");
```

Comment:
```suggestion
            using var _ = Logger.Log.OnEnterAndExit(command.ToDisplayString("text/log"));
```

It's probably easier to understand and more consistent if the formatter handles 100% of the formatting.

Reply:
There's a PocketLogger (and Microsoft.Extensions.Logging) perf concern with passing a string that can change over time here. We can discuss whether to address that here. You didn't introduce an issue. It was already present here.

Reply:
Good call. I'll removed the `Command:` here prefix in favor of the type name prefix that is already logged as part of `ToDisplayString`.

Surprisingly though, the above middleware registration does not seem to work. I couldn't find any `Command:` prefixes in the log even without the changes in the current PR. I haven't had a chance to investigate why yet - but I am also unsure whether we gain much by logging commands via the above middleware registration (on the root composite kernel) when the scheduler is already logging all the commands it sees. Would it be ok to remove the above middleware registration completely?

Reply:
Are you starting the tool with `--verbose`?

Reply:
Ah my bad. Looks like `LogCommandsToPocketLogger` is only called from within tests (unlike - so please ignore what I said above.